add post token test and add .vscode gitignore

y2468101216 opened the pull request:
## 📝 相關的 issue / Related Issues
- 主 issue #36 
- 實作 issue #45 

## ⛏ 變更內容 / Details of Changes
- 增加 route_token_test.go
- 增加 gitignore .vscode

完成了 route_token_test.go 的測試部分，使用 mock

ars0915 committed:
feat: search board articles by author or title or both

ars0915 committed:
modify: search case insensitive

ars0915 committed:
fix: query parameter to lower

Commit:
add search articles condition: recommendCountGe, modify get query param
method

y2468101216 committed:
add post token test and add .vscode gitignore

mkfsn commented on route_token_test.go:
```diff
@@ -0,0 +1,50 @@
+package main
```
想請教一下，這個檔案是有討論過要放在哪邊嗎？因為是 package main 然後是 Test 讓我有點困惑。

sageofcross replied:
依照 Go 的慣例，測試檔案通常與被測試檔案放在一起，唯一區別是文件是否跟著 `_test` 後綴。

而 `package` 的命名你會看見兩種，這邊以 `fmt` 當例子：

- fmt
- fmt_test

其中的差別在於前者因為處於同一個 `package` 所以可以使用 unexported func ；後者則是視為不同套件所以只能測試 exported func ，也就是白相與黑箱測試。

為了統一建議使用黑箱（也就是 `fmt_test` ）來當測試的套件名，正常情況下其實不該測 Private 的函式。

y2468101216 replied:
這樣的話我要順便把原有的 route_user_test.go改掉嗎

teyuchang commented:
Could you please rebase onto the latest `development` branch?
We have changed the project layout in this PR https://github.com/Ptt-official-app/Ptt-backend/pull/23

y2468101216 commented:
> Could you please rebase onto the latest `development` branch?
> We have changed the project layout in this PR #23

ok，I will rebase on Monday.

ars0915 committed:
Merge branch 'development' into feat/search-articles

ars0915 committed:
fix: mock func parameter

ars0915 committed:
fix: file name typo

ars0915 committed:
Add delivery test

y2468101216 commented:
> Could you please rebase onto the latest `development` branch?
> We have changed the project layout in this PR #23

reabse done.  if I need to squash, please tell me

ars0915 committed:
Merge branch 'development' into feat/search-articles

ars0915 committed:
modify: mock part of route boards articles test

ars0915 committed:
move http mock to seperate file

ars0915 committed:
Add usecase board article test and move mock to seperate file

ars0915 committed:
fix golint

PichuChen commented on internal/delivery/http/route_users_test.go:
```diff
@@ -1,6 +1,8 @@
 package http
 
 import (
+	"time"
```
為什麼要另外把 "time" package 移動到上面兩行？

PichuChen commented on internal/mock/user_record.go:
```diff
@@ -0,0 +1,43 @@
+package mock
```
mock package 應該要被拿掉了

ars0915 committed:
Add recommend_count_le param and recommend_count cond is set flag

ars0915 committed:
Modify url value set at test

ars0915 committed:
Rename article search condition parameters, modify search condition for
readable

ars0915 committed:
Add test case at get board articles

ars0915 committed:
modify search article condition for readable

ars0915 committed:
fix: nil slice to empty slice for json return [] instead of null

PichuChen committed:
Merge pull request #24 from ars0915/feat/search-articles

Feat/search articles

gn01842919 committed:
feat: implement missing fields of user record

Some fields are missing in the current implementation
since go-bbs does not export those methods yet.

Use a mock to fill the fields

gn01842919 committed:
add error as return value for Repository.GetUsers()

PichuChen committed:
Merge pull request #69 from gn01842919/feature/user_infomation

Implement issue #58 [主線] [PTT] 實作個人看板的使用者資訊 (repository)

PichuChen commented:
LGTM
